### DISCUSSION TITLE
Page cached to 404 (return notFound)

### DISCUSSION THREAD

**Discussion by sragneau** [vercel/next.js#53445] [Help]
### Summary

Hi,

In this Next.js code, we have a caching issue where a page has already cached version with static file, and we need to get a 404 page because the content has been deleted into our CMS
It seems that the return notFound does'nt generate a new static file. 

How can we ensure that the 404 page is displayed instead of the cached page?

### Additional information

```js
export default function Page() {
    return (
     <Page></Page>
    )
  }
  
  export async function getStaticProps(context) {
    const path = getPathServer(context)
    const [erreur, data] = await getData(path)
    const codeErreur = getCodeError(erreur)
    if (codeErreur) {
      if (codeErreur === 'PAGE_404') {
        return { notFound: true }
      }
      throw erreur
    }
     
  
    return {
      props: {
        data, 
      },
      revalidate: 5,
    }
  }
  
  export async function getStaticPaths() {
    return {
      paths: [],
      fallback: 'blocking',
    }
  }
```

**Comment by Uy575** [comment]
Next Js provide custom error page which detect automatically wether error page needs to show or not, before next 13 you can make page with name 404.js it will render automatically and in next 13 you can create error.js page which with auto render when there is an error in page.
may be you are refering to this.

**Reply by sragneau** [reply]
Thanks Uy575, we've done that, and it works if we try to get a page that has never existed. But when a cached previous version is found, the 404 page doesn't work. In fact, if we return 'not found,' the system takes the previous cached page, but our need is to display a 404 page because this page has been deleted from our CMS.

We have this problem in build start.
For information our next.config is set with the option isrMemoryCacheSize like that :
experimental: {
    esmExternals: false,
    isrMemoryCacheSize: 0,
  },

**Reply by Uy575** [reply]
If you have previously cached the old version of the deleted page, you may want to clear the Next.js build cache to ensure that the updated behavior takes effect immediately. You can do this by deleting the .next directory in your project, and then running next build and next start again to regenerate the static pages and rebuild the cache.

With these steps, when a user accesses a deleted page, Next.js will serve the dynamic 404 page with a proper 404 status code, and it won't be cached, ensuring that the user sees the updated "Page Not Found" message instead of the cached version of the deleted page.


Addittionally you can add 
export async function getStaticProps() {
  return {
    notFound: true,
  };
}
under not found page by setting notFound: true, you're indicating that this page should return a 404 status code. This will prevent Next.js from statically generating and caching the 404 page.
